gcsweb/cmd/gcsweb/gcsweb.go

```diff
@@ -452,18 +460,52 @@ func (s *server) gcsRequest(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-// splitBucketObject breaks a path into the first part (the bucket), and
-// everything else (the object).
-func splitBucketObject(path string) (string, string) {
+func bucketWithScheme(prowPath *prowv1.ProwPath) string {
+	return fmt.Sprintf("%s://%s", prowPath.StorageProvider(), prowPath.Bucket())
+}
+
+func providerPrefix(provider string) string {
```

Comment:
Seems like `providerPrefix` and `providerName` would be useful logic for the `provider` package - wouldn't every consumer need these?

Reply:
Regarding `providerPrefix`: on the contrary, the function here rewrites `gs` to `gcs` for backward-compatibility. This is specific to gcsweb. All other places probably use the `gs` path prefix, consistent with `providers.GS` and the `gs://` scheme. I would keep this function close to gcsweb, as we probably don't want any other component to do a similar (inconsistent) rewriting.

Regarding `providerName`: I couldn't find another place that uses a provider name description similar to the page title of gcsweb. I don't know if the function is really worth moving.

Reply:
Gotcha, didn't realize this is some weird bespoke compatibility thing. Regarding the display-name - let's attach it directly to the provider constants or add a helper in the `provider` package, even if we're the only consumer. Seems less like a GCSWeb-ism and more of a `provider`-ism.

Reply:
Alright, works for me. Done :)